packages/client/src/Call.ts

```diff
@@ -2000,28 +2002,36 @@ export class Call {
    * Applicable only for ringing calls.
    */
   private scheduleAutoDrop = () => {
-    clearTimeout(this.dropTimeout);
-    this.leaveCallHooks.add(
-      createSubscription(this.state.settings$, (settings) => {
-        if (!settings) return;
-        // ignore if the call is not ringing
-        if (this.state.callingState !== CallingState.RINGING) return;
-
-        const timeoutInMs = this.isCreatedByMe
-          ? settings.ring.auto_cancel_timeout_ms
-          : settings.ring.incoming_call_timeout_ms;
-
-        // 0 means no auto-drop
-        if (timeoutInMs <= 0) return;
+    this.cancelAutoDrop();
+
+    const settings = this.state.settings;
+    if (!settings) return;
+    // ignore if the call is not ringing
+    if (this.state.callingState !== CallingState.RINGING) return;
+
+    const timeoutInMs = this.isCreatedByMe
+      ? settings.ring.auto_cancel_timeout_ms
+      : settings.ring.incoming_call_timeout_ms;
+
+    // 0 means no auto-drop
+    if (timeoutInMs <= 0) return;
+
+    this.dropTimeout = setTimeout(() => {
+      // the call might have stopped ringing by this point,
+      // e.g. it was already accepted and joined
+      if (this.state.callingState !== CallingState.RINGING) return;
```

Comment:
An additional guard to prevent dropping a call that is no longer `RINGING` (e.g. was accepted or rejected on the current device/tab).